pgmpy/factors/Factor.py

```diff
@@ -191,39 +191,26 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(),variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        sum_variable = np.zeros(len(factor.cardinality))
+        sum_variable[marginalize_index] = -1
+        marg_factor = []
+        for i in itertools.product(*[range(index) for index in factor.cardinality[np.where(assign!=-1)[0]]]):
+            temp_sum = 0
+            sum_variable[sum_variable!=-1] = i
+            for index in factor._index_for_assignment(sum_variable):
+                temp_sum += factor.values[index]
```

Comment:
@palashahuja Can't we replace this loop with something like `temp_sum = np.sum(factor.values[factor._index_for_assignment(sum_variable)])` ??